src/orion/core/worker/multi_task_algo.py

```diff
@@ -0,0 +1,551 @@
+
+        These experiments and their trials were fetched from some knowledge base, and
+        are believed to be somewhat similar to the current on-going experiment.
+
+        It is the responsability of the Algorithm to implement this method in order to
+        take advantage of these points.
+
+        Parameters
+        ----------
+        warm_start_trials : Dict[Mapping, List[Trial]]
+            Dictionary mapping from ExperimentInfo objects (dataclasses containing the
+            experiment config) to the list of Trials associated with that experiment.
+        """
+        raise NotImplementedError(
+            f"Algorithm of type {type(self)} isn't warm-starteable yet."
+        )
+
+    # @contextmanager
+    # @abstractmethod
+    # def warm_start_mode(self):
+    #     pass
+
+
+import pkg_resources
+from pkg_resources import EntryPoint
+from orion.algo.base import OptimizationAlgorithm
+from functools import lru_cache
+from functools import singledispatch
+
+
+@lru_cache()
+def get_all_algos() -> Dict[str, Type[OptimizationAlgorithm]]:
+    """ Retrieve a dictionary mapping from name to algorithm class.
+    
+    NOTE: Caching this because it's relatively expensive to perform, and because the
+    available algos can't possibly change over the course of a run (as far as I can tell).
+    """
+    algos = {}
+    for entry_point in pkg_resources.iter_entry_points("OptimizationAlgorithm"):
+        entry_point: EntryPoint
+        algo_class = entry_point.load()
+        log.debug(
+            f"Found a {entry_point.name} from distribution: "
+            f"{entry_point.dist.project_name}={entry_point.dist.version}"
+        )
+        algos[entry_point.name] = algo_class
+    return algos
+
+
+@singledispatch
+def is_warmstarteable(
+    algo: Union[BaseAlgorithm, Type[BaseAlgorithm], Dict[str, Any]]
+) -> bool:
+    """Returns wether the given algo, algo type, or algo config, supports warm-starting.
+
+    Parameters
+    ----------
+    algo : Union[BaseAlgorithm, Type[BaseAlgorithm], Dict[str, Any]]
+        Algorithm instance, type of algorithm, or algorithm configuration dictionary.
+
+    Returns
+    -------
+    bool
+        Wether the input is or describes a warm-starteable algorithm.
+    """
+    return False
+
+
+@is_warmstarteable.register(type)
+def _(algo: Type[BaseAlgorithm]) -> bool:
+    return issubclass(algo, WarmStarteable)
+
+
+@is_warmstarteable.register(BaseAlgorithm)
+def _(algo: BaseAlgorithm) -> bool:
+    return isinstance(algo, WarmStarteable)
+
+
+@is_warmstarteable.register(AlgoWrapper)
+def _(algo: AlgoWrapper) -> bool:
+    # NOTE: Not directing directly to algo.unwrapped, because there might eventually be
+    # a Wrapper that makes an algo WarmStarteable. Recurse down the chain of wrappers
+    # instead.
+    return is_warmstarteable(algo.algorithm)
+
+
+@is_warmstarteable.register(str)
+def _(algo: str) -> bool:
+    available_algos = get_all_algos()
```

Comment:
Ah, that's why you implemented get_all_algos. This function should be part of the Factory in that case.

Reply:
In any case, we should not spend time on this as the Factory will be soon removed in favor of the concept class. Let's wait for this part. The PR won't be merge before the Concept class is released in v0.2.0.